tools/codegen/core/gen_settings_ids.py

```diff
@@ -0,0 +1,182 @@
+#!/usr/bin/env python2.7
```

Comment:
We don't currently. Generally there are a few manual steps and we rely on people getting this right. It's actually turned out ok so far.